### DISCUSSION TITLE
How can I use Three Dimensional Array

### DISCUSSION THREAD

**Discussion by silvercall** [beehive-lab/TornadoVM#556] [Q&A] [answered]
Hi,
I want to use three dimensional array e.g. int[][][] then store an array on third dimension
int[][][] dim = new int[5][5][120]

dim[0][0][0] = 4;
dim[0][0][1] = 44;
dim[0][0][2] = 12;

after set the values I want to start parallel loop

        for (@Parallel int i = 0; i < result[i].length; i++) {
            for (int j = 0; j < result[i][j].length; j++) {
                
            }
        }

is this somehow possible with TornadoVm. I have analysed every possible types like matrix, vector but I couldn't find any solution to solve this issue.

Can someone please help me?

**Comment by jjfumero** [comment, marked as answer]
TornadoVM does not support explicit 2D and 3D Java ND-Arrays. Instead, TornadoVM offers 2D and 3D Panama segments (implemented in with flatten arrays). I don't think we have for `int` types, but we have for `float`:

https://github.com/beehive-lab/TornadoVM/blob/e74c574880ecd3690d9659dfdc786136f81001dd/tornado-unittests/src/main/java/uk/ac/manchester/tornado/unittests/matrices/TestMatrixTypes.java#L614-L644

We can add a 3D `int`.

**Comment by silvercall** [comment]
Hi @jjfumero ,
thank you very much for the quick answer.
I will try to solve the issue with panama segments
